src/lib/orionld/troe/pgConnectionGet.cpp

```diff
@@ -92,22 +92,44 @@ PgConnection* pgConnectionGet(const char* db)
   // Search for a free but already connected PgConnection in the pool
   for (int ix = 0; ix < poolP->items; ix++)
   {
-    PgConnection* cP = poolP->connectionV[ix];
+      PgConnection* cP = poolP->connectionV[ix];
 
-    if ((cP == NULL) || (cP->busy == true))
-      continue;
+      if ((cP == NULL) || (cP->busy == true))
+          continue;
 
-    if (cP->connectionP != NULL)
-    {
-      // Great - found a free and already connected item - let's use it !
-      cP->busy = true;
-
-      sem_post(&poolP->poolSem);
-      sem_post(&poolP->queueSem);
-
-      cP->uses += 1;
-      return cP;
-    }
+      if (cP->connectionP != NULL)
+      {
+          // check if we are still connected
+          ConnStatusType pgStatus = PQstatus(cP->connectionP);
+          if (pgStatus != CONNECTION_OK)
+          {
+              LM_W(("Connection of item %d is lost, trying to re-connect...", ix));
+              // try to re-connect
+              PQreset(cP->connectionP);
+              // get status again
+              pgStatus = PQstatus(cP->connectionP);
+
+              // if still no connection
+              if (pgStatus != CONNECTION_OK)
+              {
+                  // we free this pointer that it can be used in the next call of pgConnectionGet
+                  free(poolP->connectionV[ix]);
+                  poolP->connectionV[ix] = NULL;
+                  LM_W(("Connection failed, pointer of item %d was re-set to NULL (%p)", ix, poolP->connectionV[ix]));
+                  // this time no success finding a connection that is working, try in the next loop
+                  continue;
```

Comment:
You seem to have changed the indentation to 4 spaces.
I'd really prefer to have two spaces, as in the rest of the code base.

Reply:
I changed the indentation to 2 spaces.